packages/provider-utils/src/test/unified-test-server.ts

```diff
@@ -56,14 +56,33 @@ export type UrlHandlers<
 };
 
 class TestServerCall {
-  constructor(private request: Request) {}
+  private requestClone: Request;
+
+  constructor(private request: Request) {
+    this.requestClone = request.clone();
+  }
 
   get requestBody() {
-    return this.request!.text().then(JSON.parse);
+    return this.requestClone.text().then(text => {
+      // Check if the request is multipart/form-data
+      const contentType = this.requestClone.headers.get('content-type') || '';
+      if (contentType.startsWith('multipart/form-data')) {
+        // For multipart/form-data, return the form data entries as an object
+        return this.request.formData().then(formData => {
+          const obj: Record<string, any> = {};
+          formData.forEach((value, key) => {
+            obj[key] = value;
+          });
+          return obj;
+        });
+      }
+      // For JSON requests, parse as JSON
+      return JSON.parse(text);
+    });
```

Comment:
would prefer separate helper `multipartRequestBody` instead

Reply:
(and maybe we copy the other one and change it to `jsonRequestBody` in a separate PR)